src/main/java/com/coveros/selenified/application/Check.java

```diff
@@ -0,0 +1,815 @@
+     * @param expectedCookieValue - the expected value of the cookie
+     * @param waitFor             - if waiting, how long to wait for (set to 0 if no wait is desired)
+     * @param timeTook            - the amount of time it took for wait for something (assuming we had to wait)
+     * @return String: if the cookie is present, the value of the cookie, and null if the cookie isn't present
+     */
+    default String checkCookieEquals(String cookieName, String expectedCookieValue, double waitFor, double timeTook) {
+        //record the action
+        getOutputFile().recordAction(
+                "to find cookie with the name <b>" + cookieName + VALUE_OF + expectedCookieValue + STORED, waitFor);
+        // check for the object to be present
+        String cookieValue;
+        boolean isCookiePresent = getApp().is().cookiePresent(cookieName);
+        if (isCookiePresent) {
+            cookieValue = getApp().get().cookieValue(cookieName);
+        } else {
+            getOutputFile().recordActual(COOKIE + cookieName + NOT_STORED, timeTook, Success.FAIL);
+            return "";
+        }
+        if (!cookieValue.equals(expectedCookieValue)) {
+            getOutputFile().recordActual(
+                    COOKIE + cookieName + "</b> is stored for the page, but the value of the cookie is " + cookieValue,
+                    timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(COOKIE + cookieName + VALUE_OF + cookieValue + STORED, timeTook, Success.PASS);
+        }
+        return cookieValue;
+    }
+
+    /**
+     * Checks that a cookies with the provided name has a value matching the
+     * expected value pattern. This information will be logged and recorded, with a
+     * screenshot for traceability and added debugging support.
+     *
+     * @param cookieName            - the name of the cookie
+     * @param expectedCookiePattern - the expected value of the cookie
+     */
+    void cookieMatches(String cookieName, String expectedCookiePattern);
+
+    /**
+     * Checks that a cookies with the provided name has a value matching the
+     * expected value pattern. This information will be logged and recorded, with a
+     * screenshot for traceability and added debugging support. Handles the actual
+     * logging for the verify, assert, and waitFor implementations
+     *
+     * @param cookieName            - the name of the cookie
+     * @param expectedCookiePattern - the expected value of the cookie
+     * @param waitFor               - if waiting, how long to wait for (set to 0 if no wait is desired)
+     * @param timeTook              - the amount of time it took for wait for something (assuming we had to wait)
+     * @return String: if the cookie is present, the value of the cookie, and null if the cookie isn't present
+     */
+    default String checkCookieMatches(String cookieName, String expectedCookiePattern, double waitFor, double timeTook) {
```

Comment:
getCookie should return the value of the cookie. checkCookieMatches should return a boolean.